gcloud-java-examples/src/main/java/com/google/cloud/examples/nio/ParallelCountBytes.java

```diff
@@ -0,0 +1,174 @@
+import java.io.IOException;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.MessageDigest;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * ParallelCountBytes will read through the whole file given as input.
+ *
+ * <p>This example shows how to go through all the contents of a file,
+ * in order, using multithreaded NIO reads.It also reports how long it took.
+ *
+ * <p>See the README for compilation instructions. Run this code with
+ * {@code target/appassembler/bin/ParallelCountBytes <file>}
+ */
+public class ParallelCountBytes {
+
+  private class BufWithLock {
+    public Object lock;
+    public ByteBuffer buf;
+    public boolean full;
+    public Thread t;
+
+    public BufWithLock(int size) {
+      this.buf = ByteBuffer.allocate(size);
+      this.lock = new Object();
+    }
+  }
+
+  /**
+   * See the class documentation.
+   */
+  public static void main(String[] args) throws IOException {
+    new ParallelCountBytes().start(args);
+  }
+
+  public void start(String[] args) throws IOException {
+    if (args.length == 0 || args[0].equals("--help")) {
+      help();
+      return;
+    }
+    for (String a : args) {
+      countFile(a);
+    }
+  }
+
+  private void stridedRead(SeekableByteChannel chan, int blockSize, int firstBlock, int stride, BufWithLock output) {
+    try {
+      // stagger the threads a little bit.
+      Thread.sleep(250 * firstBlock);
+      long pos = firstBlock * blockSize;
+      synchronized(output.lock) {
+        while (true) {
+          if (pos > chan.size()) {
+            break;
+          }
+          chan.position(pos);
+          // read until buffer is full, or EOF
+          while (chan.read(output.buf) > 0) {};
+          output.full = true;
+          output.lock.notifyAll();
+          if (output.buf.hasRemaining()) {
+            break;
+          }
+          // wait for main thread to process it
+          while (output.full) {
+            output.lock.wait();
+          }
+          output.buf.flip();
+          pos += stride * blockSize;
+        }
+      }
+    } catch (InterruptedException | IOException o) {
+      // this simple example doesn't handle errors, sorry.
+    }
+  }
+
+  /**
+   * Print the length of the indicated file.
+   *
+   * <p>This uses the normal Java NIO Api, so it can take advantage of any installed
+   * NIO Filesystem provider without any extra effort.
+   */
+  private void countFile(String fname) throws IOException{
+    // large buffers pay off
+    final int bufSize = 50 * 1024 * 1024;
+    try {
+      Path path = Paths.get(new URI(fname));
+      long size = Files.size(path);
+      System.out.println(fname + ": " + size + " bytes.");
+      ByteBuffer buf = ByteBuffer.allocate(bufSize);
+      int nBlocks = (int)Math.ceil( size / (double)bufSize);
```

Comment:
Space is needed between the cast indication and the casted value, no space is needed after open `(`:
```java
int nBlocks = (int) Math.ceil(size / (double) bufSize);
```